src/backend/libs/efiling-cso-client/src/main/java/ca/bc/gov/open/jag/efilingcsoclient/CsoAccountServiceImpl.java

```diff
@@ -158,7 +157,7 @@ private Client setCreateAccountClientDetails(CreateAccountRequest createAccountR
         XMLGregorianCalendar date =  CsoHelpers.date2XMLGregorian(new Date());
 
         client.setAuthenticatedClientGuid(CsoHelpers.formatUserGuid(createAccountRequest.getUniversalId()));
-        client.setAuthoritativePartyId(BigDecimal.ONE);
+        client.setAuthoritativePartyId(Keys.IDENTITY_PROVIDERS.get(createAccountRequest.getIdentityProvider().toUpperCase()));
```

Comment:
you might want to throw an exception early in the method

Reply:
👍